std/math.d

```diff
@@ -238,13 +238,19 @@ version (StdUnittest) private
         int ix;
         int iy;
         version (CRuntime_Microsoft)
+        {
             alias real_t = double;
+            string fmt = "g";
+        }
         else
+        {
             alias real_t = real;
+            string fmt = "Lg";
+        }
 
         () @trusted {
-            ix = sprintf(bufx.ptr, "%.*Lg", ndigits, cast(real_t) x);
-            iy = sprintf(bufy.ptr, "%.*Lg", ndigits, cast(real_t) y);
+            ix = sprintf(bufx.ptr, "%.*" ~ fmt, ndigits, cast(real_t) x);
+            iy = sprintf(bufy.ptr, "%.*" ~ fmt, ndigits, cast(real_t) y);
```

Comment:
`fmt` being a runtime variable means this is not guaranteed to be nul terminated. Also its not a literal so cannot be implicitly cast to `const(char)*`.